[DOCU-2804] Konnect audit logging (#5055)
Kong/docs.konghq.com: app/konnect/org-management/audit-logging/index.md

---
title: Audit Logging in Konnect
badge: enterprise
content_type: explanation
beta: true
---

Audit logging enables administrators to better spot security risks and maintain compliance of their core infrastructure. 

> No system is completely secure. Audit logs provide a mechanism to catch abnormal events when those events occur. 
> If potential security incidents are not caught early, perpetrators can cause issues or even downtime with a company’s infrastructure, 
> leading to serious consequences such as outages and lost revenue.

Reviewer: ```suggestion
Audit logs can help you detect and respond to potential security incidents when they occur. Monitoring audit logs proactively can reduce the risk of outages and end ensure continuous service for your users.  No system can ever be completely secure, but audit logs can be a key step in your incident prevention infrastructure

```

By tracking {{site.konnect_short_name}} audit logs, you gain the following benefits:

> * **Security**: System events can be used to show abnormalities to be investigated, forensic information related to breaches, or provide evidence.

Reviewer: ```suggestion
* **Security**: System events can be used to show abnormalities to be investigated, forensic information related to breaches, or provide evidence for compliance and regulatory purposes.
```

> * **Compliance**: Some regulations require audit logs, or audit logs are needed to show auditors who confirm whether certain certification standards are met.

Reviewer: ```suggestion
* **Compliance**: Regulators and auditors may require audit logs to confirm whether certain certification standards are met.
```

> * **Debugging**: Audit logs give insights into why there is a current level of efficiency or performance. 

Reviewer: ```suggestion
* **Debugging**: Audit logs can help pinpoint the root cause of efficiency or performance issues.
```

Author: Went with "determine" over "pinpoint" to avoid colloquialisms.

By tracking {{site.konnect_short_name}} audit logs, you gain the following benefits:
* **Security**: System events can be used to show abnormalities to be investigated, forensic information related to breaches, or provide evidence for compliance and regulatory purposes.
* **Compliance**: Regulators and auditors may require audit logs to confirm whether certain certification standards are met.
* **Debugging**: Audit logs can help determine the root causes of efficiency or performance issues.
* **Risk management**: Prevent issues or catch them early.

## Setting up audit logging in {{site.konnect_short_name}}

{{site.konnect_short_name}} administrators can track streams of security events and operational changes per organization.
You can do this by [setting up a webhook](/konnect/org-management/audit-logging/webhook/) to send data to any 
log collection service that supports [ArcSight CEF Format](https://docs.centrify.com/Content/IntegrationContent/SIEM/arcsight-cef/arcsight-cef-format.htm) or JSON-formatted data.

Audit logging webhooks can be configured through the {{site.konenct_short_name}} **Organization** menu, or

> using the [Organization Settings API](https://developer.konghq.com/spec/e46e7742-befb-49b1-9bf1-7cbe477ab818/d36126ee-ab8d-47b2-960f-5703da22cced).

Reviewer 2: ```suggestion
using the [Audit Logs API](https://developer.konghq.com/spec/e46e7742-befb-49b1-9bf1-7cbe477ab818/d36126ee-ab8d-47b2-960f-5703da22cced).
```

Only {{site.konnect_short_name}} organization administrators can configure and view audit log webhooks. 

![Audit log webhook](/assets/images/docs/konnect/konnect-audit-log-webhook.png)

Audit information includes authentication attempts, authorization requests, and gateway access logs. 
Each of the audit events contains a trace ID that allows events to be correlated to specific actions. 
See the [audit log reference](/konnect/org-management/audit-logging/reference/) for details on what is logged.

{:.note}
> **Note:** You can't customize the events that {{site.konnect_short_name}} sends to the logs.

## More information
* [Set up an audit log webhook](/konnect/org-management/audit-logging/webhook/)
* [Set up an audit log replay job](/konnect/org-management/audit-logging/replay-job/)
* [Audit log event reference](/konnect/org-management/audit-logging/reference/)

> * [Organization Settings API](https://developer.konghq.com/spec/e46e7742-befb-49b1-9bf1-7cbe477ab818/d36126ee-ab8d-47b2-960f-5703da22cced/)

Reviewer 2: ```suggestion
* [Audit Logs API](https://developer.konghq.com/spec/e46e7742-befb-49b1-9bf1-7cbe477ab818/d36126ee-ab8d-47b2-960f-5703da22cced/)
```
